### PR TITLE
Added Patient Information

### PR DESCRIPTION
**Pull request by NotSalH**
Readded all the patient information

### CODE REVIEW

File: src/main/java/medicaldoctor/controllers/UserLookupController.java [hunk 1]
```diff
@@ -100,6 +100,7 @@ public void initialize(URL location, ResourceBundle resources) {
         
         table.setRowFactory(tv -> {
             TableRow<User> row = new TableRow<>();
+            
             row.setOnMouseClicked(event -> {
```
Comment (NotSalH) [inline src/main/java/medicaldoctor/controllers/UserLookupController.java:104]: I tried making your suggestion with the throws Exception but you can't overload the init method .

Comment (tlacasse) [reply src/main/java/medicaldoctor/controllers/UserLookupController.java:104]: you dont overload, you just add the throws declaration

File: src/main/java/medicaldoctor/controllers/NavigationBarController.java [hunk 2]
```diff
@@ -195,21 +195,21 @@ public void initialize(URL location, ResourceBundle resources) {
...
-        //buttonMustHaveThisPermission(buttonWhiteBloodCell, Permission.VIEW_HEMOTOLOGIC_BUTTONS);
-        //buttonMustHaveThisPermission(buttonLiverFunction, Permission.VIEW_HEMOTOLOGIC_BUTTONS);
-        //buttonMustHaveThisPermission(buttonRenalFunction, Permission.VIEW_HEMOTOLOGIC_BUTTONS);
-        //buttonMustHaveThisPermission(buttonElectrolyteTest, Permission.VIEW_HEMOTOLOGIC_BUTTONS);
-        //buttonMustHaveThisPermission(buttonUrinaryTest, Permission.VIEW_HEMOTOLOGIC_BUTTONS);
-        //buttonMustHaveThisPermission(buttonStoolTest, Permission.VIEW_HEMOTOLOGIC_BUTTONS);
-        //buttonMustHaveThisPermission(buttonXRay, Permission.VIEW_RADIOLOGIC_BUTTONS);
-        //buttonMustHaveThisPermission(buttonCT, Permission.VIEW_RADIOLOGIC_BUTTONS);
-        //buttonMustHaveThisPermission(buttonMRI, Permission.VIEW_RADIOLOGIC_BUTTONS);
+        buttonMustHaveThisPermission(buttonLabPatients, Permission.ACCESS_ALL_LAB_PATIENTS);
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/controllers/NavigationBarController.java]: we still want to hide the unused buttons

File: src/main/java/medicaldoctor/controllers/PatientRegistrationController.java [hunk 3]
```diff
@@ -65,20 +76,177 @@
...
+           Patient patient = new Patient();
+           PatientVisit visit = new PatientVisit();
+           PatientPrescription prescription = new PatientPrescription();
+           makePatient(patient, visit, prescription);
+        }
     }
-
+    
+   void makePatient(Patient patient, PatientVisit visit, PatientPrescription prescription) {
+        patient.setFirstName(textFirstName.getText());
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/controllers/PatientRegistrationController.java]: this is where we hook up the service

File: src/main/resources/NavigationBar.fxml [hunk 4]
```diff
@@ -14,16 +14,40 @@
...
+      <Button fx:id="buttonRedBloodCell" mnemonicParsing="false" onAction="#redBloodCell" prefHeight="42.0" prefWidth="94.0" text="Red Blood Cell" />
+      <Button fx:id="buttonWhiteBloodCell" mnemonicParsing="false" onAction="#whiteBloodCell" prefHeight="42.0" prefWidth="105.0" text="White Blood Cell" />
+      <Button fx:id="buttonLiverFunction" mnemonicParsing="false" onAction="#liverFunctions" prefHeight="42.0" prefWidth="90.0" text="Liver Function" />
+      <Button fx:id="buttonRenalFunction" mnemonicParsing="false" onAction="#renalFuctions" prefHeight="42.0" prefWidth="101.0" text="Renal Function" />
+      <Button fx:id="buttonElectrolyteTest" mnemonicParsing="false" onAction="#electrolyteTest" prefHeight="42.0" prefWidth="101.0" text="Electrolyte Test" />
+      <Button fx:id="buttonUrinaryTest" mnemonicParsing="false" onAction="#urinaryTest" prefHeight="42.0" prefWidth="97.0" text="Urinary Test" />
+      <Button fx:id="buttonStoolTest" mnemonicParsing="false" onAction="#stoolTest" prefHeight="42.0" prefWidth="90.0" text="Stool Test" />
```
Comment (tlacasse) [inline src/main/resources/NavigationBar.fxml]: we still want to hide the unused buttons

**Review by tlacasse — CHANGES_REQUESTED** [review]

File: src/main/java/medicaldoctor/controllers/DoctorSearchController.java [hunk 5]
```diff
@@ -80,7 +79,7 @@ public void initialize(URL location, ResourceBundle resources) {
                     AppSession.setPatientSelection(rowData);
                     AppSession.setPatientFlag(1);
                     try {
-                        //AppSession.CONTROLLER_MANAGER.loadAndShowScreen(LookUp.PATIENT_PROFILE);
+                        AppSession.CONTROLLER_MANAGER.loadAndShowScreen(LookUp.PATIENT_PROFILE);
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/controllers/DoctorSearchController.java:82]: patient on doctor search

File: src/main/java/medicaldoctor/controllers/NavigationBarController.java [hunk 6]
```diff
@@ -195,7 +195,7 @@ public void initialize(URL location, ResourceBundle resources) {
         buttonMustHaveThisPermission(buttonPatientLookup, Permission.ACCESS_PATIENT_LOOKUP);
         buttonMustHaveThisPermission(buttonDoctorLookup, Permission.ACCESS_DOCTOR_LOOKUP);
         buttonMustHaveThisPermission(buttonUserLookup, Permission.ACCESS_USER_LOOKUP);
-        //buttonMustHaveThisPermission(buttonMyLabRequests, Permission.ACCESS_MY_LAB_REQUESTS);
+        buttonMustHaveThisPermission(buttonMyLabRequests, Permission.ACCESS_MY_LAB_REQUESTS);
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/controllers/NavigationBarController.java]: we dont have lab requests

File: src/main/java/medicaldoctor/controllers/PatientLookupController.java [hunk 7]
```diff
@@ -114,15 +102,27 @@ public void initialize(URL location, ResourceBundle resources){
...
         
+        try (DatabaseScope scope = new DatabaseScope()) {
+            doctors = User.getAll();
+        } catch (Exception ex) {
+           throw new InitializeException(ex);
+        }
+        
+        for(int i = 0; i < doctors.size(); i++){
+            if(doctors.get(i).getType().getDashboardName().equals("DoctorDashboard")){
+                doctor_choicebox.getItems().add(doctors.get(i).getFullName());
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/controllers/PatientLookupController.java:113]: doctor in patientlookup

**Review by tlacasse — CHANGES_REQUESTED** [review]

File: src/main/java/medicaldoctor/controllers/PatientRegistrationController.java [hunk 8]
```diff
@@ -65,20 +78,157 @@
...
+        }
     }
-
+    
+   private void makePatient() throws Exception{
+        patient.firstName = textFirstName.getText();
+        patient.lastName = textLastName.getText() ;
+        patient.age = Integer.parseInt(isInteger(ageNumber.getText()));
+        patient.sex =textSex.getText();
+        patient.dateOfBirth = LocalDate.of(1997, 10, 1);
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/controllers/PatientRegistrationController.java]: hardcoded

File: src/main/java/medicaldoctor/controllers/PatientRegistrationController.java [hunk 9]
```diff
@@ -65,20 +78,157 @@
...
+        patient.addressCity =(textStreetMailingAddress.getText());
+        patient.addressCity = (textCityMailingAddress.getText());
+        patient.addressState =((String) stateMailingAddressChoiceBox.getValue());
+        patient.addressZipCode = (zipCodeMailingAddressNumber.getText());
+        patient.billingAddressStreet = (textStreetBillingAddress.getText());
+        patient.billingAddressCity = (textCityBillingAddress.getText());
+        patient.billingAddressState = ((String) stateBillingAddressChoiceBox.getValue());
+        patient.billingAddressZipCode = (zipCodeBillingAddressNumber.getText());
+        patient.primaryDoctor = (user_hash.get((String) doctor_dropdown.getValue()));
+        patient.socialSecurityNumber = ("123-56-7890");
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/controllers/PatientRegistrationController.java]: hardcoded

File: src/main/resources/NavigationBar.fxml [hunk 10]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (tlacasse) [inline src/main/resources/NavigationBar.fxml]: extra line at beginning

File: src/main/resources/NavigationBar.fxml [hunk 11]
```diff
@@ -25,4 +26,4 @@
          </image>
       </ImageView>
   </items>
-</ToolBar>
+</ToolBar>
```
Comment (tlacasse) [inline src/main/resources/NavigationBar.fxml]: no empty line at end

**Review by tlacasse — CHANGES_REQUESTED** [review]

**Pull request merged**